core/domain/user_services.py

```diff
@@ -723,11 +769,23 @@ def get_user_settings_by_gae_id(gae_id, strict=False):
     Raises:
         Exception. The value of strict is True and given gae_id does not exist.
     """
-    user_auth_details_model = user_models.UserAuthDetailsModel.get_by_auth_id(
-        feconf.AUTH_METHOD_GAE, gae_id)
-    if user_auth_details_model is not None:
+    user_id = None
+    gae_id_to_user_id_model = (
+        user_models.GaeIdToUserIdModel.get(gae_id, strict=False))
+    # If the GaeIdToUserIdModels are not yet generated use UserAuthDetailsModel
```

Comment:
There should be a TODO to delete this part of the code completely after the job is run, right?

Reply:
Done.